src/models/analytics-query.ts

```diff
@@ -0,0 +1,85 @@
+import Joi from 'joi';
+
+const constraintSchema = Joi.object().keys({
+  field: Joi.alternatives().try(Joi.string(), Joi.number()).required(),
+  comparison: Joi.string().required(),
+  value: Joi.alternatives().try(Joi.string().allow(''), Joi.number(), Joi.boolean()),
+});
+
+const conditionSchema = Joi.object().keys({
+  type: Joi.string().valid('or', 'and').required(),
+  restrictions: Joi.array().items(Joi.alternatives().try(
+    Joi.link('#analyticsQueryCondition').required(),
+    constraintSchema.required(),
+  )).required(),
+}).id('analyticsQueryCondition');
+
+const schema = Joi.object().keys({
+  source: Joi.string().required(),
+  columns: Joi.array().items(Joi.alternatives().try(
+    Joi.string().required(),
+    Joi.object().keys({ type: Joi.string().valid('count', 'share').required(), condition: conditionSchema }).required(),
+    Joi.object().keys({ type: Joi.string().valid('sum', 'mean', 'min', 'max').required(), condition: conditionSchema, field: Joi.string().required() }).required(),
+    Joi.object().keys({ type: Joi.string().valid('timeGroup').required(), field: Joi.string().required(), granularity: Joi.string().required() }).required(),
+  )).required(),
+  filter: conditionSchema,
+  offset: Joi.string(),
+  sort: Joi.array().items(Joi.object().keys({
+    columnIndex: Joi.number().required(),
+    descending: Joi.boolean().required(),
+  })).default(() => []),
+  rowsPerPage: Joi.number().default(20).min(10).max(100),
+})
+  .description('A geographical object representing a point. Is used to connect one or more single segment lines.')
+  .tag('node');
+
+type TimeGranularity =
+  'day'
+  | 'week'
+  | 'month'
+  | 'year'
+  | 'dayOfTheWeek'
+  | 'monthOfTheYear';
+
+type Comparison = '=' | '>=' | '<=' | '>' | '<' | 'like' | '<>' | 'isNull' | 'isNotNull';
```

Comment:
i think should be also added `notLike` comparison

Reply:
Yeah, good point. I won't add it now (as the SDK has already progressed a few versions further, for which the platform has not yet caught up), but will put it on my todo!